### DISCUSSION TITLE
How to store which features a code was compiled with

### DISCUSSION THREAD

**Discussion by ahkole** [aiidateam/aiida-core#6213] [Q&A]
Many DFT codes have optional features that are only enabled when compiled with a certain library. I.e., there might be support for reading and writing netCDF files only if compiled with the optional netCDF library. For one of the codes I am working with it would be preferable to read info from netCDF files if they are available and also to provide the information for new calculation (for example info from a previous calculation to restart a new one) in the netCDF format if the code supports this. Is there any (recommended) way in AiiDA right now to store the enabled features of a certain code that can be used by a Calcjob to know if it's possible to use the netCDF format for providing input when starting a calculation? For extracting info from netCDF files after a finished calculation you could simply check if these files are present in the output folder and if so read from those and if not default to other files, but for supplying information to a calculation you still have to start I think it's more complicated. Many codes will probably crash if you try to supply them with netCDF files but netCDF support was not compiled in. Any help/suggestions for this would be appreciated.

**Comment by zhubonan** [comment]
I think usually the code is explicitly supplied for each calculation/workflow rather than automatic selected. You can add some info in the label of the Code to help, like `mycode-netcdf-v-xx-xx`. 

One can also add metadata in the `extras` after a code is created, then let the Calculation check it before submission.
